plugins/telemetry/telemetry.h

```diff
@@ -6,48 +6,90 @@ namespace dronecore {
-        double longitude_deg;
-        float absolute_altitude_m;
-        float relative_altitude_m;
+        double latitude_deg; /**< latitude from -90 to +90 in degrees */
+        double longitude_deg; /**< longitude from -180 to 180 in degrees */
+        float absolute_altitude_m; /**< altitude AMSL (above mean sea level in meters */
+        float relative_altitude_m; /**< altitude relative to takeoff altitude in meters */
     };
 
+    /**
+     * Quaternion type
+     *
+     * All rotations and axis systems follow the right-hand rule.
+     * The Hamilton quaternion product definition is used.
+     * A zero-rotation quaternion is represented by (1,0,0,0).
+     * The quaternion could also be written as w + xi + yj + zk.
+     */
     struct Quaternion {
-        float w;
-        float x;
-        float y;
-        float z;
+        float w; /**< quaternion entry 0 also denoted as a */
+        float x; /**< quaternion entry 1 also denoted as b */
+        float y; /**< quaternion entry 2 also denoted as c */
+        float z; /**< quaternion entry 3 also denoted as d */
     };
 
+    /**
+     * Euler angle type
+     *
+     * All rotations and axis systems follow the right-hand rule.
+     * The Euler angles follow the convention of a 3-2-1 intrinsic Tait-Bryan rotation sequence.
+     */
     struct EulerAngle {
-        float roll_deg;
-        float pitch_deg;
-        float yaw_deg;
+        float roll_deg; /**< roll angle in degrees, positive is banking to the right */
+        float pitch_deg; /**< pitch angle in degrees, positive is pitching nose up. */
+        float yaw_deg; /**< yaw angle in degrees, positive is clock-wise seen from above. */
     };
 
+    /**
+     * Ground speed type
+     *
+     * The ground speed is represented in the NED (North East Down) frame and in meters/second.
+     */
     struct GroundSpeedNED {
-        float velocity_north_m_s;
-        float velocity_east_m_s;
-        float velocity_down_m_s;
+        float velocity_north_m_s; /**< velocity in North direction in meters/second. */
+        float velocity_east_m_s; /**< velocity in East direction in meters/second. */
+        float velocity_down_m_s; /**< velocity in Down direction in meters/second. */
     };
 
+    /**
+     * GPS information type
+     */
     struct GPSInfo {
-        int num_satellites;
-        int fix_type;
+        int num_satellites; /**< Number of visible satellites used for solution */
+        int fix_type; /**< Fix type (0: no GPS, 1: no fix, 2: 2D fix, 3: 3D fix, 4: DGPS fix,
+                                     5: RTK float, 6: RTK fixed) */
     };
 
+    /**
+     * Battery type
+     */
     struct Battery {
-        float voltage_v;
-        float remaining_percent;
+        float voltage_v; /**< Voltage in volts */
+        float remaining_percent; /**< Estimated battery percentage remaining from 0.0 to 1.0 */
     };
 
+    /**
+     * Flight modes
+     */
     enum class FlightMode {
```

Comment:
Docstrings? Ideally with link to the official docs for mode in user guide.

Reply:
Put a link.